CSharp/Library/Microsoft.Bot.Builder/Luis/LuisService.cs

```diff
@@ -248,75 +248,86 @@ public static void Fix(LuisResult result)
                 if (result.TopScoringIntent != null)
                 {
                     result.Intents = new List<IntentRecommendation> { result.TopScoringIntent };
-                }
-            }
-        }
-
-        async Task<LuisResult> ILuisService.QueryAsync(Uri uri, CancellationToken token)
-        {
-            string json;
-            using (var client = new HttpClient())
-            using (var response = await client.GetAsync(uri, HttpCompletionOption.ResponseContentRead, token))
-            {
-                response.EnsureSuccessStatusCode();
-                json = await response.Content.ReadAsStringAsync();
-            }
-
-            try
-            {
-                var result = JsonConvert.DeserializeObject<LuisResult>(json);
-                Fix(result);
-                return result;
-            }
-            catch (JsonException ex)
-            {
-                throw new ArgumentException("Unable to deserialize the LUIS response.", ex);
-            }
-        }
-    }
-
-    /// <summary>
-    /// LUIS extension methods.
-    /// </summary>
-    public static partial class Extensions
-    {
-        /// <summary>
-        /// Query the LUIS service using this text.
-        /// </summary>
-        /// <param name="service">LUIS service.</param>
-        /// <param name="text">The query text.</param>
-        /// <param name="token">The cancellation token.</param>
-        /// <returns>The LUIS result.</returns>
-        public static async Task<LuisResult> QueryAsync(this ILuisService service, string text, CancellationToken token)
-        {
-            var luisRequest = service.ModifyRequest(new LuisRequest(query: text));
-            return await service.QueryAsync(luisRequest, token);
-        }
-
+                }
+            }
+        }
+
+        public void ApplyThreshold(LuisResult result)
+        {
+            if (result.TopScoringIntent.Score > model.Threshold)

```

Comment:
>TopScoringIntent [](start = 23, length = 16)

is this ever null?  i.e. are intents only included in Intents if verbose is true?

Reply:
provides the same behavior with verbose true or false